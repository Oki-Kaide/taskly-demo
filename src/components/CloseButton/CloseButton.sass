.close-button
  background-color: #eceff8
  border-radius: 100%
  height: 35px
  width: 35px
  cursor: pointer
  display: flex
  flex-direction: row
  justify-content: center
  align-items: center