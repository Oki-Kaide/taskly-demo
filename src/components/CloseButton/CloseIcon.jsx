import React from 'react';
import PropTypes from 'prop-types';

const CloseIcon = ({
  fill, width, opacity, className, viewBox,
}) => (
  <svg
    width={width}
    height={width}
    viewBox={viewBox}
    fill={fill}
    opacity={opacity}
    className={`svg-icon ${className}`}
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z" />
  </svg>
);

CloseIcon.propTypes = {
  fill: PropTypes.string,
  width: PropTypes.string,
  opacity: PropTypes.string,
  className: PropTypes.string,
  viewBox: PropTypes.string,
};

CloseIcon.defaultProps = {
  fill: '#000',
  width: '14px',
  opacity: '1.0',
  className: '',
  viewBox: '0 0 14 14',
};

export default CloseIcon;
