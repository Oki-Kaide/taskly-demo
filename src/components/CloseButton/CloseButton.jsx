import React from 'react';
import PropTypes from 'prop-types';
import CloseIcon from './CloseIcon';
import './CloseButton.sass';

const CloseButton = ({handleOnClick}) => {
  return(
    <div className="close-button" onClick={handleOnClick}>
      <CloseIcon />
    </div>
  )
}

export default CloseButton;

CloseButton.propTypes = {
  handleOnClick: PropTypes.func,
};

CloseButton.defaultProps = {
  handleOnClick: () => {},
}