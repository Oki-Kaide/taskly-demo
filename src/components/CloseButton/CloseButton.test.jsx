import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CloseButton from './CloseButton';
import CloseIcon from './CloseIcon';

const createTestProps = (props) => ({
  handleOnClick: jest.fn(),
  ...props,
});

describe('CloseButton component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<CloseButton {...props} />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<CloseButton {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('tests click', () => {
    const component = shallow(<CloseButton />);
    component.find('div').simulate('click');
  });
});

describe('CloseIcon component', () => {
  it('renders without crashing', () => {
    shallow(<CloseIcon />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<CloseIcon />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
