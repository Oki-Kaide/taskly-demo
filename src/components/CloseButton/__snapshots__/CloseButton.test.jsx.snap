// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton component matches the rendered snapshot 1`] = `
<div
  className="close-button"
  onClick={[MockFunction]}
>
  <CloseIcon
    className=""
    fill="#000"
    opacity="1.0"
    viewBox="0 0 14 14"
    width="14px"
  />
</div>
`;

exports[`CloseIcon component matches the rendered snapshot 1`] = `
<svg
  className="svg-icon "
  fill="#000"
  height="14px"
  opacity="1.0"
  version="1.1"
  viewBox="0 0 14 14"
  width="14px"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z"
  />
</svg>
`;
